lib/talkingscoreslib.py

```diff
@@ -324,6 +553,114 @@ def group_chord_pitches_by_octave(self, chord):
 
         return chord_pitches_by_octave
 
+    
+    #for all / selected / unselected
+    def generate_midi_filename_sel(self, prefix, range_start=None, range_end=None, output_path="", sel=""):
+        base_filename = os.path.splitext(os.path.basename(self.filepath))[0]
+        if (range_start!=None):
+            midi_filename = os.path.join(output_path, ("%s.mid?sel=%s&start=%d&end=%d&t=100&c=n" % ( base_filename, sel, range_start,range_end )) )
```

Comment:
Have a look at Python f-strings for this sort of thing https://realpython.com/python-f-strings/

Reply:
Ah yes - I use something similar in React - it is much easier to read

Reply:
Fixed in 5fceb81
Thank you - f-strings do greatly improves readability!